packages/contracts/deploy/2_new_version/22_publish.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  METADATA,
+  PLUGIN_CONTRACT_NAME,
+  PLUGIN_REPO_ENS_NAME,
+  PLUGIN_SETUP_CONTRACT_NAME,
+  VERSION,
+} from '../../plugin-settings';
+import {addCreatedVersion, getAragonDeploymentsInfo} from '../../utils/helpers';
+import {
+  PLUGIN_REPO_PERMISSIONS,
+  toHex,
+  uploadToIPFS,
+} from '@aragon/osx-commons-sdk';
+import {PluginRepo__factory, PluginSetup__factory} from '@aragon/osx-ethers';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import path from 'path';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  console.log(
+    `Publishing ${PLUGIN_SETUP_CONTRACT_NAME} as v${VERSION.release}.${VERSION.build} in the "${PLUGIN_REPO_ENS_NAME}" plugin repo`
+  );
+
+  const {deployments} = hre;
+  const [deployer] = await hre.ethers.getSigners();
+
+  // Upload the metadata to IPFS
+  const releaseMetadataURI = `ipfs://${await uploadToIPFS(
+    JSON.stringify(METADATA.release, null, 2)
+  )}`;
+  const buildMetadataURI = `ipfs://${await uploadToIPFS(
+    JSON.stringify(METADATA.build, null, 2)
+  )}`;
+
+  console.log(`Uploaded release metadata: ${releaseMetadataURI}`);
+  console.log(`Uploaded build metadata: ${buildMetadataURI}`);
+
+  // Get PluginSetup
+  const setup = await deployments.get(PLUGIN_SETUP_CONTRACT_NAME);
+
+  // Get PluginRepo
+  const pluginRepo = PluginRepo__factory.connect(
+    getAragonDeploymentsInfo(hre.network.name)[hre.network.name].address,
+    deployer
+  );
+
+  // Check release number
+  const latestRelease = await pluginRepo.latestRelease();
+  if (VERSION.release > latestRelease + 1) {
+    throw Error(
+      `Publishing with release number ${VERSION.release} is not possible. 
+        The latest release is ${latestRelease} and the next release you can publish is release number ${
+        latestRelease + 1
+      }.`
+    );
+  }
+
+  // Check build number
+  const latestBuild = (await pluginRepo.buildCount(VERSION.release)).toNumber();
+  if (VERSION.build < latestBuild) {
+    throw Error(
+      `Publishing with build number ${VERSION.build} is not possible. The latest build is ${latestBuild}. Aborting publication...`
+    );
+  }
+  if (VERSION.build > latestBuild + 1) {
+    throw Error(
+      `Publishing with build number ${VERSION.build} is not possible. 
+        The latest build is ${latestBuild} and the next release you can publish is release number ${
+        latestBuild + 1
+      }. Aborting publication...`
+    );
+  }
+
+  // Create Version
+  let tx;
+  if (
+    await pluginRepo.callStatic.isGranted(
+      pluginRepo.address,
+      deployer.address,
+      PLUGIN_REPO_PERMISSIONS.MAINTAINER_PERMISSION_ID,
+      '0x00' // TODO
```

Comment:
Confirm TODO can be left in commit

Reply:
My bad, fixed in 6a7151a8c0eab555f3ca1cb17d43f7c95e415f6c